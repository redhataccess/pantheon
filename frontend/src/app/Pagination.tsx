import React, { Fragment } from 'react';
import {Button,Badge,Level, LevelItem} from '@patternfly/react-core';

export interface IProps{
    handleMoveLeft: Function
    handleMoveRight: Function
    pageNumber : number
    nextPageRecordCount: number
    noOfRecordsOnPage: number
}

export class Pagination extends React.Component<IProps> {

private fetchBlocks = () => {
    const totalBlocks = 2;
    const PREVIOUS_PAGE = 'PREVIOUS';
    const NEXT_PAGE = 'NEXT';
    const pages = [PREVIOUS_PAGE,NEXT_PAGE,totalBlocks];
    return pages;
}

render() {

    const blocks = this.fetchBlocks();
    return (
      <Fragment>
        <Level gutter="md">
          <LevelItem>
            <nav aria-label="Countries Pagination">
              {blocks.map((page, index) => {
                if (page === "PREVIOUS") return (
                  this.props.pageNumber === 1 ?
                    <Button isDisabled href="#" target="_blank" variant="primary" onClick={() => this.props.handleMoveLeft()}>Previous</Button>
                    :
                    <Button href="#" variant="primary" onClick={() => this.props.handleMoveLeft()}>Previous</Button>
                );

                if (page === "NEXT") return (
                  this.props.nextPageRecordCount === 0 ?
                    <Button isDisabled href="#" target="_blank" variant="secondary" onClick={() => this.props.handleMoveRight()}>Next</Button>
                    :
                    <Button href="#" variant="secondary" onClick={() => this.props.handleMoveRight()}>Next</Button>
                );
                
                return (
                  <Badge>Page No: {this.props.pageNumber}</Badge>
                );
              })}
            </nav>
          </LevelItem>
          <LevelItem>
            <Badge>No. of records on page: {this.props.noOfRecordsOnPage}</Badge>
          </LevelItem>
        </Level>
      </Fragment>
    );

  }
} 

export default IProps;