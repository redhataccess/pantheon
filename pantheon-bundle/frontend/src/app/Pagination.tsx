import React, { Fragment } from "react";
import { DropdownItem, Level, LevelItem } from "@patternfly/react-core";
import { ContextSelector, ContextSelectorItem } from "@patternfly/react-core";
import { Dropdown } from "@app/Dropdown";

export interface IProps {
  handleMoveLeft: () => any
  handleMoveRight: () => any
  handleMoveToFirst: () => any
  handlePerPageLimit: (itemsPerPage) => any
  handleItemsPerPage: (itemsPerPage) => any
  pageNumber: number
  nextPageRecordCount: number
  perPageLimit: number
  showDropdownOptions: boolean
  bottom: boolean
  className?: string
  currentBulkOperation: string
}

class Pagination extends React.Component<IProps> {

  public dropdownItems = [
    "5 items per page",
    "25 items per page",
    "50 items per page",
    "100 items per page"
  ];

  public state = {
    filteredItems: this.dropdownItems,
    isExpanded: false,
    isOpen: false,
    renderSearch: false,
    searchValue: "",
    selected: this.dropdownItems[0]

  };

  public firstPageButton = true;
  public previousPageButton = true;
  public nextPageButton = true;

  public render() {
    const { isOpen, filteredItems } = this.state;

    if (this.props.pageNumber === 1 && this.props.nextPageRecordCount !== 0) {
      this.firstPageButton = true;
      this.previousPageButton = true;
      this.nextPageButton = false;
    }
    if (this.props.nextPageRecordCount === 0 && this.props.pageNumber !== 1) {
      this.firstPageButton = false;
      this.previousPageButton = false;
      this.nextPageButton = true;
    }
    if (this.props.pageNumber !== 1 && this.props.nextPageRecordCount !== 0) {
      this.firstPageButton = false;
      this.previousPageButton = false;
      this.nextPageButton = false;
    }
    if (this.props.pageNumber === 1 && this.props.nextPageRecordCount === 0) {
      this.firstPageButton = true;
      this.previousPageButton = true;
      this.nextPageButton = true;
    }
    if(this.props.currentBulkOperation !== ""){
      this.firstPageButton = true;
      this.previousPageButton = true;
      this.nextPageButton = true;
    }

    return (
      <Fragment>
        <Level>
          <LevelItem />
          <LevelItem />
          <LevelItem>
            <nav aria-label="Pagination" className={this.props.className}>
              <div>
                <div>
                  <div className="example">
                    <div className="ws-preview">
                      <div className="pf-c-pagination" id="pagination-options-menu-top">
                        <div className="pf-c-dropdown">
                          <div className="pf-c-options-menu__toggle pf-m-plain pf-m-text">
                            {!this.props.showDropdownOptions && !this.props.bottom &&
                              <ContextSelector
                                isOpen={isOpen}
                                onToggle={this.onToggle}
                              />}
                            {this.props.showDropdownOptions &&
                              <Dropdown
                                perPageValue={this.dropDownValue}
                                newPerPagevalue={this.props.perPageLimit + " items per page"}
                              />
                            }
                            <button disabled={this.firstPageButton} data-action="first" aria-label="Go to first page" className="pf-c-button pf-m-plain" type="button" onClick={this.props.handleMoveToFirst}><svg fill="currentColor" height="1em" width="1em" viewBox="0 0 448 512" aria-hidden="true" role="img">
                              <path d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z" transform="" /></svg></button>
                            <button disabled={this.previousPageButton} data-action="previous" aria-label="Go to previous page" className="pf-c-button pf-m-plain" type="button" onClick={this.props.handleMoveLeft}><svg fill="currentColor" height="1em" width="1em" viewBox="0 0 256 512" aria-hidden="true" role="img">
                              <path d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z" transform="" /></svg></button>
                            Page: {this.props.pageNumber}
                            <button disabled={this.nextPageButton} data-action="next" aria-label="Go to next page" className="pf-c-button pf-m-plain" type="button" onClick={this.props.handleMoveRight}><svg fill="currentColor" height="1em" width="1em" viewBox="0 0 256 512" aria-hidden="true" role="img">
                              <path d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z" transform="" /></svg></button>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div></div>
              </div>
            </nav>
          </LevelItem>
        </Level>
      </Fragment>
    );
  }

  private onToggle = isOpen => {
    this.setState({
      isOpen: !this.state.isOpen
    });
  };

  private dropDownValue = (value) => {
    this.setState({
      isOpen: !this.state.isOpen,
      itemsPerPage: Number(value.substr(0, value.indexOf(" "))),
      selected: value
    }, () => {
      this.props.handlePerPageLimit(Number(value.substr(0, value.indexOf(" "))))
      this.props.handleItemsPerPage(Number(value.substr(0, value.indexOf(" "))))
    });
  };
}

export { Pagination }