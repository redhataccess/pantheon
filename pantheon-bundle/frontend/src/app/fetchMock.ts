if (!self.fetch) {
    self.fetch = (input: RequestInfo, init?: RequestInit | undefined) => {
        const p = new Promise<Response>((resolve, reject) => {
            // just don't do anything. this exists so that our tests don't try and actually resolve fetches,
            // (which is exactly what isomorphic-fetch tried to do while we were importing it...)
            // and calling nothing here allows us to skip all of our fetch.then() and fetch.catch() code.
        })
        return p
    }
}