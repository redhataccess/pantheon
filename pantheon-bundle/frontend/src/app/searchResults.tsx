import React, { Component } from "react";

import {
  Table,
  TableHeader,
  TableBody,
  headerCol,
} from "@patternfly/react-table";
import { Checkbox } from '@patternfly/react-core';
import { Tooltip } from '@patternfly/react-core';
import "@app/app.css";
import styles from "@patternfly/react-styles/css/components/Table/table";
import { Pagination } from "@app/Pagination"
import {
  Divider,
  EmptyState,
  EmptyStateBody,
  Title,
  EmptyStateIcon,
  EmptyStateVariant
} from "@patternfly/react-core";
import { SearchIcon, ExclamationTriangleIcon, ExclamationCircleIcon } from "@patternfly/react-icons";
import CheckCircleIcon from "@patternfly/react-icons/dist/js/icons/check-circle-icon"
import { SlingTypesPrefixes } from "./Constants";

export interface IProps {
  contentType: string
  keyWord: string
  filters: { ctype: any, status: any }
  productsSelected: string[]
  repositoriesSelected: string[]
  userAuthenticated: boolean
  onGetdocumentsSelected: (documentsSelected) => any
  onSelectContentType: (contentType) => any
  currentBulkOperation: string
  disabledClassname: string
  bulkOperationCompleted: boolean
}
export interface ISearchState {

  columns: [
    { title: string },
    { title: string, cellTransforms: any },
    { title: string },
    { title: string },
    { title: string }
  ],
  displayLoadIcon: boolean
  // filterQuery: string
  isEmptyResults: boolean
  isSearchException: boolean
  // states for pagination
  nextPageRowCount: number
  page: number
  pageLimit: number
  itemsPerPage: number
  results: any
  rows: any
  canSelectAll: boolean
  showDropdownOptions: boolean
  bottom: boolean
}
class SearchResults extends Component<IProps, ISearchState> {

  constructor(props) {
    super(props);
    this.state = {
      // states for table
      columns: [
        { title: "" },
        { title: "Title", cellTransforms: [headerCol()] },
        { title: "Repository" },
        { title: "Upload Date" },
        { title: "Last Published Date" }
      ],
      displayLoadIcon: true,
      // filterQuery: "",
      isEmptyResults: false,
      isSearchException: false,
      // states for pagination
      nextPageRowCount: 1,
      page: 1,
      pageLimit: 5,
      itemsPerPage: 5,
      results: [
        {
          "pant:transientPath": "",
          "pant:dateUploaded": "",
          "name": "",
          "jcr:title": "",
          "jcr:description": "",
          "productVersion": "",
          "sling:transientSource": "",
          "pant:transientSourceName": "",
          "checkedItem": false,
          "publishedDate": "-",
          "pant:moduleType": "-",
          "variant": "",
          "validations": ""
        }
      ],
      // states for table
      rows: [
        {
          cells: ["", "", "", ""]
        }
      ],
      canSelectAll: true,
      showDropdownOptions: true,
      bottom: true,
    };

    this.onSelect = this.onSelect.bind(this);
    this.toggleSelect = this.toggleSelect.bind(this);
  }

  public componentDidMount() {
    this.doSearch()
  }

  public componentDidUpdate(prevProps) {
    if (this.props.repositoriesSelected !== prevProps.repositoriesSelected
      || this.props.productsSelected !== prevProps.productsSelected
      || this.props.keyWord !== prevProps.keyWord
      || this.props.filters !== prevProps.filters
      || this.props.onGetdocumentsSelected !== prevProps.onGetdocumentsSelected
      || (this.props.bulkOperationCompleted !== prevProps.bulkOperationCompleted
        && this.props.currentBulkOperation.length === 0
      )
    ) {
      this.doSearch()
    }
  }

  public render() {
    const { columns, rows, canSelectAll, results } = this.state;

    return (
      <React.Fragment>
        {!this.state.isEmptyResults &&
          <div className={this.props.disabledClassname}>
            <Checkbox
              label="Can select all"
              className="pf-u-mb-lg"
              isChecked={canSelectAll}
              onChange={this.toggleSelect}
              aria-label="toggle select all checkbox"
              id={"toggle-select-all-" + this.props.contentType}
              name={"toggle-select-all-" + this.props.contentType}
            />
            <Table
              onSelect={this.onSelect}
              canSelectAll={canSelectAll}
              aria-label={"Selectable Table " + this.props.contentType}
              cells={columns}
              rows={rows}
            >
              <TableHeader className={styles.modifiers.nowrap} />
              <TableBody className="results__table-body" />
            </Table></div>}

        {!this.state.isEmptyResults && <Pagination
          handleMoveLeft={this.updatePageCounter("L")}
          handleMoveRight={this.updatePageCounter("R")}
          handleMoveToFirst={this.updatePageCounter("F")}
          pageNumber={this.state.page}
          nextPageRecordCount={this.state.nextPageRowCount}
          handlePerPageLimit={this.changePerPageLimit}
          handleItemsPerPage={this.changePerPageLimit}
          perPageLimit={this.state.pageLimit}
          showDropdownOptions={this.state.showDropdownOptions}
          bottom={this.state.bottom}
          className="results__pagination"
          currentBulkOperation={this.props.currentBulkOperation}
          
        />}

        {this.state.isEmptyResults && <EmptyState variant={EmptyStateVariant.small} className="search-results--empty">
          <EmptyStateIcon icon={SearchIcon} />
          <Title headingLevel="h2" size="lg">
            No results found
        </Title>
          <EmptyStateBody>
            No results match the filter criteria. Select repo filter to show results.
        </EmptyStateBody>

        </EmptyState>
        }
        <Divider />
      </React.Fragment>
    );
  }

  // private methods
  private buildSearchQuery() {
    if ((this.props.repositoriesSelected && this.props.repositoriesSelected.length > 0)
      || (this.props.productsSelected && this.props.productsSelected.length > 0)) {
      let backend = "/pantheon/internal/modules.json?"

      if (this.props.repositoriesSelected) {
        this.props.repositoriesSelected.map(repo => {
          if (backend.endsWith("?") || backend.endsWith("&")) {
            backend += "repo=" + repo
          } else {
            backend += "&repo=" + repo
          }
        })
      }

      //TODO: enable product filter in the query
      // if (this.props.productsSelected) {
      //   console.log("[productsSelected]", this.props.productsSelected)
      //   this.props.productsSelected.map(product => {
      //     backend += "product=" + product + "&"
      //   })
      // }

      if (this.props.contentType) {
        backend += "&ctype=" + this.props.contentType
      }

      if (this.props.keyWord) {
        backend += "&search=" + this.props.keyWord.trim()
      }

      if (this.props.filters) {
        if (this.props.filters.ctype) {
          this.props.filters.ctype.map(type => {
            backend += "&type=" + type
          })
        }
        if (this.props.filters.status) {
          this.props.filters.status.map(stat => {
            backend += "&status=" + stat
          })
        }
      }

      backend += "&offset=" + ((this.state.page - 1) * this.state.itemsPerPage) + "&limit=" + this.state.itemsPerPage

      if (!backend.includes("Updated") && !backend.includes("direction")) {
        backend += "&key=Updated&direction=desc"
      }

      return backend
    } else {
      this.setState({ isEmptyResults: true })
      return ""
    }
  }

  // private setFilterQuery = (filterQuery: string) => {
  //   this.setState({ filterQuery })
  // };

  // methods for pagination
  private updatePageCounter = (direction: string) => () => {
    if (direction === "L" && this.state.page > 1) {
      this.setState({ page: this.state.page - 1 }, () => {
        this.doSearch()
      })
    } else if (direction === "R") {
      this.setState({ page: this.state.page + 1 }, () => {
        this.doSearch()
      })
    } else if (direction === "F") {
      this.setState({ page: 1 }, () => {
        this.doSearch()
      })
    }
  }

  // Handle gateway timeout on slow connections.
  private doSearch = () => {
    this.setState({ displayLoadIcon: true })
    if (this.buildSearchQuery()) {
      fetch(this.buildSearchQuery())
        .then(response => response.json())
        .then(responseJSON => {
          this.setState({ results: responseJSON.results, nextPageRowCount: responseJSON.hasNextPage ? 1 : 0 })
          const data = new Array()

          responseJSON.results.map((item, key) => {
            const lastUpdateDate = item["pant:publishedDate"]

            let docIcon = lastUpdateDate !== "-" ? <div><Tooltip position="top" content={<div>Published successfully</div>}><CheckCircleIcon className="p2-search__check-circle-icon" /></Tooltip></div> : null
            if (docIcon === null) {
              const productVersion = item["productVersion"] != undefined ? item["productVersion"] : "-"

              docIcon = productVersion == "-" ? <div><Tooltip position="top" content={<div>Metadata missing</div>}><ExclamationTriangleIcon color="#f0ab00" /></Tooltip></div> : null
            }
            if (item.validations !== undefined && item.validations.length > 1) {
              let vIcon = <div><Tooltip position="top" content={<div>{item.validations}</div>}><ExclamationCircleIcon color="#c9190b" /></Tooltip></div>
              docIcon = <span>{docIcon}{vIcon}</span>
            }
            const cellItem = new Array()
            cellItem.push(docIcon)
            if (this.props.userAuthenticated) {
              cellItem.push({ title: <a href={"/pantheon/#" + item["sling:resourceType"].substring(SlingTypesPrefixes.PANTHEON.length) + "/" + item['pant:transientPath'] + "?variant=" + item.variant}> {item["jcr:title"] !== "-" ? item["jcr:title"] : item["pant:transientPath"]} </a> })
            } else {
              let docTitle = item["jcr:title"] !== "-" ? item["jcr:title"] : item["pant:transientPath"]
              cellItem.push(docTitle)
            }
            { title: "Last Published Date" }
            cellItem.push(item["pant:transientSourceName"])
            cellItem.push(item["pant:dateUploaded"])
            cellItem.push(lastUpdateDate)

            data.push({ cells: cellItem })
          })
          if (responseJSON.results.length > 0) {
            this.setState({ rows: data })
          } else {
            const rows = [{ cells: ["", "", "", ""] }]
            this.setState({ rows })
          }
        })
        .then(() => {
          if (JSON.stringify(this.state.results) === "[]") {
            this.setState({
              displayLoadIcon: false,
              isEmptyResults: true,
              // selectAllCheckValue: false
            })
          } else {
            this.setState({
              displayLoadIcon: false,
              isEmptyResults: false,
              // selectAllCheckValue: false,
            })
          }
        })
        .catch(error => {
          // might be a timeout error
          this.setState({
            displayLoadIcon: false,
            isSearchException: true
          }, () => { console.log("[doSearch] error ", error) })

        })
    } else {
      this.setState({ isEmptyResults: true })
    }
  }

  public changePerPageLimit = (pageLimitValue) => {
    this.setState({ pageLimit: pageLimitValue, page: 1, itemsPerPage: pageLimitValue }, () => {
      this.props.onGetdocumentsSelected([])
      this.doSearch()
      this.props.onSelectContentType("")
    })
  }

  private onSelect(event, isSelected, rowId) {
    let rows;
    if (rowId === -1) {
      rows = this.state.rows.map(oneRow => {
        oneRow.selected = isSelected;
        return oneRow;
      });
    } else {
      rows = [...this.state.rows];
      rows[rowId].selected = isSelected;
    }
    this.setState({
      rows
    });
    // update props.documentSelected
    let selectedRows;
    selectedRows = this.state.rows.map(oneRow => {
      if (oneRow.selected === true) {
        return oneRow;
      }
    })
    // filter undefined values
    selectedRows = selectedRows.filter(r => r !== undefined)

    this.props.onGetdocumentsSelected(selectedRows);

    if (selectedRows.length > 0 || isSelected == true) {
      this.props.onSelectContentType(this.props.contentType);
    }
    if (selectedRows.length == 0 && isSelected == false) {
      this.props.onSelectContentType("");
    }
  }

  private toggleSelect(checked) {
    this.setState({
      canSelectAll: checked
    });
  }
}


export { SearchResults }; 