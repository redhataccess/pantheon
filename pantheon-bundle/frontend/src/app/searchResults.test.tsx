import React from "react";
import { SearchResults } from "@app/searchResults";
import { shallow, mount } from "enzyme";
import sinon from "sinon"
import renderer from "react-test-renderer"
import "@app/fetchMock"
import { any } from "prop-types";
// import { props } from "@app/TestResources"
import { Button, Drawer, Toolbar, ExpandableSection, Divider, SimpleList, Select, EmptyState, Title } from "@patternfly/react-core";
import { HashRouter as Router, Link } from "react-router-dom"
import { Pagination } from "@app/Pagination"
import { Table } from "@patternfly/react-table"

const anymatch = require("anymatch")
const props = {
    contentType: "assemblies",
    keyWord: "test",
    filters: { ctype: ["PROCEDURE"], status: ["draft"] },
    productsSelected: [],
    repositoriesSelected: ["test"],
    userAuthenticated: true,
    onGetdocumentsSelected: (documentsSelected) => anymatch,
    onSelectContentType: (contentType) => anymatch,
    currentBulkOperation: "",
    disabledClassname: ""
}

const propsEmptyState = {
    contentType: "",
    keyWord: "",
    filters: { ctype: [], status: [] },
    productsSelected: [],
    repositoriesSelected: [],
    userAuthenticated: true,
    onGetdocumentsSelected: (documentsSelected) => anymatch,
    onSelectContentType: (contentType) => anymatch,
    currentBulkOperation: "",
    disabledClassname: ""

}

describe("SearchResults tests", () => {
    test("should render default Search component", () => {
        const view = shallow(<SearchResults {...props} />);
        expect(view).toMatchSnapshot();
    });

    it("should render a Table component", () => {
        const wrapper = mount(<Router><SearchResults {...props} /></Router>)
        wrapper.setState({isEmptyResults: false})
        const table = wrapper.find(Table)
        expect(table.exists()).toBe(true)
    })

    it("should render an EmptyState component", () => {
        const wrapper = mount(<Router><SearchResults {...propsEmptyState} /></Router>)
        const emptyState = wrapper.find(EmptyState)
        expect(emptyState.exists()).toBe(true)
    })

    it("should render a Title component", () => {
        const wrapper = mount(<Router><SearchResults {...propsEmptyState} /></Router>)
        const title = wrapper.find(Title)
        expect(title.exists()).toBe(true)
    })

    it("should render a Divider", () => {
        const wrapper = mount(<Router><SearchResults {...props} /></Router>)
        const divider = wrapper.find(Divider)
        expect(divider.exists()).toBe(true)
    })

});
