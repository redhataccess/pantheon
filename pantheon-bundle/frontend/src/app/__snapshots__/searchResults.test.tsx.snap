// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults tests should render default Search component 1`] = `
<Fragment>
  <div
    className=""
  >
    <Checkbox
      aria-label="toggle select all checkbox"
      className="pf-u-mb-lg"
      id="toggle-select-all-assemblies"
      isChecked={true}
      isDisabled={false}
      isValid={true}
      label="Can select all"
      name="toggle-select-all-assemblies"
      onChange={[Function]}
    />
    <Table
      aria-label="Selectable Table assemblies"
      borders={true}
      canSelectAll={true}
      cells={
        Array [
          Object {
            "title": "",
          },
          Object {
            "cellTransforms": Array [
              [Function],
            ],
            "title": "Title",
          },
          Object {
            "title": "Repository",
          },
          Object {
            "title": "Upload Date",
          },
          Object {
            "title": "Last Published Date",
          },
        ]
      }
      className=""
      contentId="expanded-content"
      dropdownDirection="down"
      dropdownPosition="right"
      expandId="expandable-toggle"
      gridBreakPoint="grid-md"
      isStickyHeader={false}
      onSelect={[Function]}
      ouiaSafe={true}
      role="grid"
      rowLabeledBy="simple-node"
      rows={
        Array [
          Object {
            "cells": Array [
              "",
              "",
              "",
              "",
            ],
          },
        ]
      }
      selectVariant="checkbox"
      variant={null}
    >
      <TableHeader
        className="pf-m-nowrap"
      />
      <Component
        className="results__table-body"
      />
    </Table>
  </div>
  <Pagination
    bottom={true}
    className="results__pagination"
    handleItemsPerPage={[Function]}
    handleMoveLeft={[Function]}
    handleMoveRight={[Function]}
    handleMoveToFirst={[Function]}
    handlePerPageLimit={[Function]}
    nextPageRecordCount={1}
    pageNumber={1}
    perPageLimit={5}
    showDropdownOptions={true}
  />
  <Divider />
</Fragment>
`;
